components/playlist/browser/resources/reducers/player_reducer.ts

```diff
@@ -30,6 +43,30 @@ const playerReducer: Reducer<PlayerState | undefined> = (
     case types.PLAYER_STOPPED_PLAYING_ITEM:
       state = { ...state, playing: false }
       break
+
+    case types.PLAYER_PLAY_NEXT_ITEM:
+      if (state.currentList && state.currentItem) {
+        const items = state.currentList.items
+        const currentIndex = items.indexOf(state.currentItem)
+        console.assert(currentIndex !== -1)
+        if (currentIndex !== items.length - 1) {
```

Comment:
I don't know if it's important, but `console.assert` doesn't throw, so if we don't find the item this will select the first item in the list `-1 + 1 === 0`

Reply:
Oh, good point. Throwing an error would be good.